eva-accession-release-automation/analyze_mapping_weight_impact/lookup_overweight_snps_in_release_files.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script looks up the impact of overweight SNPs
+
+import click
+from snpmapinfo_metadata import *
+
+
+def export_snpmapinfo_for_species(species_info, metadata_connection_handle, export_dir):
+    exported_filenames_and_assemblies = []
+    species_name = species_info["database_name"]
+
+    weight_criteria_query = "select 'rs'||trim(cast(snp_id as text)) from dbsnp_{0}.{1} where weight > 3"
+    copy_statement = "COPY ({0}) TO STDOUT WITH CSV"
+
+    with get_db_conn_for_species(species_info) as species_db_connection_handle:
+        snpmapinfo_table_names = get_snpmapinfo_table_names_for_species(species_info)
+        for snpmapinfo_table_name in snpmapinfo_table_names:
+            distinct_asm, type_of_asm = get_distinct_asm_with_overweight_snps_in_snpmapinfo_table(snpmapinfo_table_name,
+                                                                                                  species_info)
+            for asm in distinct_asm:
+                if type_of_asm == "assembly_name":
+                    specific_query = weight_criteria_query + " and assembly = '{0}'".format(asm)
+                else:
+                    specific_query = weight_criteria_query + " and asm_acc = '{0}' and asm_version = '{1}'"\
+                        .format(asm.split(".")[0], asm.split(".")[1])
+                specific_query = specific_query.format(species_name, snpmapinfo_table_name) + " order by 1"
+                try:
+                    associated_GCA_assembly = lookup_GCA_assembly(species_name, snpmapinfo_table_name, asm,
+                                                                  metadata_connection_handle)
+                except Exception as e:
+                    logger.error(e)
+                    continue
+                output_file_name = os.path.sep.join([export_dir,
```

Comment:
```suggestion
                output_file_name = os.path.join([export_dir,
```

Reply:
No, I **do mean** path separator for a given OS. **os.path.sep** captures that.

Reply:
That's what `os.path.join` does and I realised I made a mistake in the suggestion it should have been:
```python
output_file_name = os.path.sep.join(export_dir, 
                                    "{0}_{1}_{2}_overweight_snps.csv".format(species_name,
                                                                             snpmapinfo_table_name,
                                                                             associated_GCA_assembly))
```
which save you creating the list as well